apps/challenges/aws_utils.py

```diff
@@ -988,3 +984,13 @@ def challenge_workers_start_notifier(sender, instance, field_name, **kwargs):
                 )
             else:
                 construct_and_send_worker_start_mail(challenge)
+
+        if prev and not curr:
+            response = delete_workers([challenge])
+            count, failures = response["count"], response["failures"]
+            if count != 1:
```

Comment:
@KhalidRmb What is this condition checking? Can you please mention in the comment?

What if I have 5 workers for a challenge running and I unapproved a challenge.
- Would all the 5 workers will be deleted?
- Do I need to manually reduce them to 1 and then unapprove the challenge?

Reply:
@RishabhJain2018 It basically checks that the number of successful deletions is same as the number of challenges passed in the array.

Yes, all 5 workers will be deleted. (The implementation is that the workers will be scaled down to 0, and then the service is deleted.)

Reply:
okay, can you please add it as comment.